File: packages/ts/client/src/txbuilder/createMarket.ts
import { Address, IInstruction, TransactionSigner } from "@solana/kit";
import {
  CreateMarketInstructionDataArgs,
  getCreateMarketInstruction,
  getMarketAddress,
  getTunaConfigAddress,
} from "../index.ts";

export async function createMarketInstruction(
  authority: TransactionSigner,
  pool: Address,
  args: CreateMarketInstructionDataArgs,
): Promise<IInstruction> {
  const tunaConfig = (await getTunaConfigAddress())[0];
  const marketAddress = (await getMarketAddress(pool))[0];

  return getCreateMarketInstruction({
    authority,
    tunaConfig,
    market: marketAddress,
    pool,
    ...args,
  });
}

File: rust-sdk/client/src/txbuilder/create_market.rs
use crate::instructions::{CreateMarket, CreateMarketInstructionArgs};
use crate::{get_market_address, get_tuna_config_address};
use solana_program::instruction::Instruction;
use solana_program::pubkey::Pubkey;
use solana_program::system_program;

pub fn create_market_instruction(authority: &Pubkey, pool: &Pubkey, args: CreateMarketInstructionArgs) -> Instruction {
    let tuna_config_address = get_tuna_config_address().0;
    let market_address = get_market_address(pool).0;

    let ix_builder = CreateMarket {
        authority: *authority,
        tuna_config: tuna_config_address,
        market: market_address,
        pool: *pool,
        system_program: system_program::ID,
    };

    ix_builder.instruction(args)
}
